src/app/model/ChessEngine.ts

```diff
@@ -119,4 +138,44 @@ export class ChessEngine implements ChessModel {
   public possibleMoves(position: Square): Square[] {
     return this.pieceMovesNotCausingAllyKingCheckmate(position);
   }
+
+  private onKingWasChecked(event: KingWasChecked): void {
+    this.checkedKing = { kingSide: event.king.side, position: event.onSquare };
+  }
+
+  private kingWasChecked(chosenPiece: Piece): KingWasChecked | undefined {
+    const kingsSide = chosenPiece.side === Side.WHITE ? Side.BLACK : Side.WHITE;
+    const kingPosition = this.kingPosition(this.board, kingsSide);
+    if (!kingPosition) {
+      return undefined;
+    }
+    const isKingChecked = this.isKingChecked(this.board, kingsSide);
+    return isKingChecked
+      ? {
+          eventType: 'KingWasChecked',
+          king: new King(kingsSide),
+          onSquare: kingPosition,
+        }
+      : undefined;
+  }
+
+  private kingWasUnchecked(pieceWasMoved: PieceWasMoved): KingWasUnchecked | undefined {
+    if (!this.checkedKing) {
+      return undefined;
+    }
+    const kingSide =
+      pieceWasMoved.piece.side === this.checkedKing.kingSide ? pieceWasMoved.piece.side : this.changeSide(pieceWasMoved.piece.side);
+    const isKingChecked = this.isKingChecked(this.board, kingSide);
+    return !isKingChecked
+      ? {
+          eventType: 'KingWasUnchecked',
+          king: new King(pieceWasMoved.piece.side),
+          onSquare: this.checkedKing?.position as Square,
```

Comment:
Tutaj jest błąd, albo inaczej trzeba to nazwać, albo może w ogóle nam to niepotrzebne? 
Co oznacza onSquare teraz? Że król był szachowany na tym polu ale juz nie jest, czy ze teraz król przestał być szachowany i jest na tym polu? Bo mógł przestać być szachowany przez ruch królem... :P 
Teraz tak mysle, ze moze w tym evencie nam nic jest niepotrzebne oprocz eventType. 
Bo w sumie cała reszte wiemy z poprzedniego. Co sądzisz :) ?

Reply:
no właśnie miałem takie mieszane uczucie co do tego onSquare :P szczególnie, że mamy teraz pole, które przechowuje szachowane królaa a dodatkowo sytuacja w której dwa szachy wystąpią równolegle jest niemożliwa. jak dal mnie możemy zostawić tylko typ

Reply:
jestem za :) Mój błąd, że tak opisałem w zadaniu